plugins/flutter_plugins/path_provider_fde/linux/path_provider_plugin.cc

```diff
@@ -0,0 +1,184 @@
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar);
+
+  virtual ~PathProviderPlugin();
+
+ private:
+  PathProviderPlugin();
+  // Called when a method is called on plugin channel;
+  void HandleMethodCall(
+      const flutter::MethodCall<EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<EncodableValue>> result);
+};
+
+// static
+void PathProviderPlugin::RegisterWithRegistrar(
+    flutter::PluginRegistrar *registrar) {
+  auto channel = std::make_unique<flutter::MethodChannel<EncodableValue>>(
+      registrar->messenger(), "plugins.flutter.io/path_provider",
+      &flutter::StandardMethodCodec::GetInstance());
+
+  // Uses new instead of make_unique due to private constructor.
+  std::unique_ptr<PathProviderPlugin> plugin(new PathProviderPlugin());
+
+  channel->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto &call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+PathProviderPlugin::PathProviderPlugin() = default;
+
+PathProviderPlugin::~PathProviderPlugin() = default;
+
+void PathProviderPlugin::HandleMethodCall(
+    const flutter::MethodCall<EncodableValue> &method_call,
+    std::unique_ptr<flutter::MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare("getTemporaryDirectory") == 0) {
+    flutter::EncodableValue response("/tmp");
+    result->Success(&response);
+  } else if (method_call.method_name().compare(
+                 "getApplicationSupportDirectory") == 0) {
+    // Use the executable name as the subdirectory for now.
+    std::string exe_name = GetExecutableName();
+    if (exe_name.empty()) {
+      result->Error("Unable to get application name");
+      return;
+    }
+    // Get user defined data home directory
+    const char *xdg_home = getenv("XDG_DATA_HOME");
+    std::ostringstream response_stream;
+    // Defaults to XDG default data home directory
+    if (xdg_home == NULL) {
+      response_stream << getenv("HOME") << "/.local/share/" << exe_name;
+    } else {
+      std::string xdg_home_string(xdg_home);
+      // Need to ensure final forward slash so we can append the exe_name
+      if (xdg_home_string.back() != '/') {
+        xdg_home_string.push_back('/');
+      }
+      response_stream << xdg_home_string << exe_name;
+    }
+    // Ensure the directory exists or create it
+    if (!EnsureCreated(response_stream.str())) {
+      result->Error(
+          "Unable to create a folder for the application in the ~/.local/share "
+          "directory");
+      return;
+    }
+    flutter::EncodableValue response(response_stream.str());
+    result->Success(&response);
+  } else if (method_call.method_name().compare(
+                 "getApplicationDocumentsDirectory") == 0) {
+    // Get user defined documents directory
+    const char *xdg_documents_dir = getenv("XDG_DOCUMENTS_DIR");
```

Comment:
I don't think this will work. The XDG docs say that there will be in "a config file in $XDG_CONFIG_HOME/user-dirs.dirs (XDG_CONFIG_HOME defaults to ~/.config) that applications can read to find these directories". It never says that those values will be in the environment, and on my machine they aren't. Only a few higher-level things like XDG_CONFIG_HOME. You'll need to read it out of the config file.

(The fact that you need to read and parse a file may be a tipping point to where it's easier to convert path_provider to using a platform interface and then writing it in Dart.)